### PR TITLE
Templates init using --template flag

### LINKED ISSUE #204: can't run `atomkraft` without `.atomkraft` directory
The project directory logic includes `.atomkraft/config.toml` file.

https://github.com/informalsystems/atomkraft/blob/e18c3ab78f04d7431180aab82495e390cd7dddb1/atomkraft/utils/project.py#L17-L27

This creates problems when someone tries to git-clone a project and execute `atomkraft` in it.
```
You are outside of Atomkraft project
...
```

I suggest checking only for a valid `atomkraft.toml` in a directory.

### LINKED ISSUE #205: `atomkraft init` should generate a `.gitignore` file
Since an `atomkraft` project generates many artifacts, the generated project should contain an appropriate `.gitignore`.

We don't need [GitPython](https://pypi.org/project/GitPython) dependency. We can write the file at the project root.

Note that `atomkraft` still doesn't initialize the project as a git repo. The current workflow of creating a git `atomkraft` project repo is,

```
atomkraft init my_project
cd my_project
git init .
```

### LINKED ISSUE #206: `atomkraft init` should generate a hello world project instead of an empty one
Conventionally any project initialization command generates a bare minimum _hello-world_ or _2+3=5_ project.

It would be great if `atomkraft init` adopts this culture.

### PR DESCRIPTION
**Pull request by udit-gulati**
Fixes https://github.com/informalsystems/atomkraft/issues/206

Fixes https://github.com/informalsystems/atomkraft/issues/205

Fixes https://github.com/informalsystems/atomkraft/issues/204

### CODE REVIEW

**Comment by udit-gulati** [comment]
@rnbguy please have a look.

File: atomkraft/cli/__init__.py [hunk 1]
```diff
@@ -84,6 +84,7 @@ def version():
 )
 def init(
     name: str = typer.Argument(..., help="Name of new directory", show_default=False),
+    template: str = typer.Option("transfer", help="Name of project template to use"),
```
Comment (rnbguy) [inline atomkraft/cli/__init__.py:87]: `template` argument should be optional. Default value can be `transfer`.

File: atomkraft/templates/cosmwasm/cosmwasm-contract/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+[package]
+name = "counter"
+version = "0.1.0"
+authors = ["djordje.pesic <djordje.pesic@ethernal.tech>"]
```
Comment (rnbguy) [inline atomkraft/templates/cosmwasm/cosmwasm-contract/Cargo.toml]: We have to be careful with the project file parameters.

File: atomkraft/templates/cosmwasm/cosmwasm-contract/LICENSE [hunk 3]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (rnbguy) [inline atomkraft/templates/cosmwasm/cosmwasm-contract/LICENSE]: shouldn't be generated.

File: atomkraft/utils/project.py [hunk 4]
```diff
@@ -14,16 +17,34 @@ def __init__(self) -> None:
...
+                    f"Configuration file pyproject.toml not found, populating in root {cwd}"
+                )
+                # copy pyproject.toml file from template
+                copy_file_from_template("pyproject.toml", cwd)
+            if not (cwd / ATOMKRAFT_INTERNAL_DIR / "config.toml").exists():
+                logging.warn(
+                    f"Configuration file config.toml not found, populating in {cwd/ATOMKRAFT_INTERNAL_DIR}"
+                )
+                # copy config.toml file from template
+                copy_file_from_template(os.path.join(ATOMKRAFT_INTERNAL_DIR, "config.toml"), cwd)
```
Comment (rnbguy) [inline atomkraft/utils/project.py]: I would prefer a situation where these files not generated.

If a user just have an `atomkraft.toml` with their chain configuration and just wants to spawn a local testnet, they don't really need `.pyproject.toml` and `config.toml`

**Review by rnbguy — CHANGES_REQUESTED** [review]
Thanks for the PR @udit-gulati . I added some comments.

Let's start with one template first. I would suggest to remove the cosmwasm one and keep bank transfer as default.

Also, the generated templates should be concise. If you just copied from examples, maybe I need to trim them little bit.

File: atomkraft/cli/__init__.py (hunk 1 above)
Comment (udit-gulati) [reply atomkraft/cli/__init__.py:87]: `typer.Option()` does make the argument optional with "transfer" as default here.

Comment (rnbguy) [reply atomkraft/cli/__init__.py:87]: Ah yes :100: Don't know how I missed it. :sweat_smile:

**Review by rnbguy — APPROVED** [review]

**Comment by rnbguy** [comment]
Thanks @udit-gulati !

**Pull request merged**